presences/src/main/resources/public/ts/controllers/calendar.ts

```diff
@@ -389,6 +394,30 @@ export const calendarController = ng.controller('CalendarController',
                 }
             };
 
+            vm.getEventReason = (event: CourseEvent): string => {
+                switch (event.type_id) {
+                    case EventType.LATENESS:
+                        return vm.reasonsMap.get(event.reason_id) ? vm.reasonsMap.get(event.reason_id).label : '';
+                    default:
+                        return '';
+                }
+            };
+
+            vm.getEventDescription = (event: CourseEvent): string => {
+                switch (event.type_id) {
```

Comment:
On est d'accord tu risques pas d'avoir un "cannot read property of undefined" ?
Ca vaut pas le coup de le check tu penses ?